internal/nix/profile.go

```diff
@@ -230,16 +230,13 @@ func profileInstall(args *ProfileInstallArgs) error {
 	cmd := exec.Command(
 		"nix", "profile", "install",
 		"--profile", args.ProfilePath,
-		"--impure", // Needed to allow flags from environment to be used.
 		FlakeNixpkgs(args.NixpkgsCommit)+"#"+args.Package,
 	)
 	cmd.Args = append(cmd.Args, ExperimentalFlags()...)
 	if args.Priority != "" {
 		cmd.Args = append(cmd.Args, "--priority", args.Priority)
 	}
 	cmd.Args = append(cmd.Args, args.ExtraFlags...)
-
-	cmd.Env = DefaultEnv()
```

Comment:
I do think this is needed:

```
❯ devbox add slack
Warning: local and global devbox.json have different nixpkgs commits. Will use the local version. This may lead to version mismatch and nix store bloat.

Installing package: slack.

[1/1] slack
        error: Package ‘slack-4.29.149’ in /nix/store/3hzps14zspk0mkxdjjk3d85vhspibb9c-source/pkgs/applications/networking/instant-messengers/slack/default.nix:82 has an unfree license (‘unfree’), refusing to evaluate.
               a) To temporarily allow unfree packages, you can use an environment variable
                  for a single invocation of the nix tools.
                    $ export NIXPKGS_ALLOW_UNFREE=1
                Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
                (Flake) command, `--impure` must be passed in order to read this
                environment variable.
               b) For `nixos-rebuild` you can set
                 { nixpkgs.config.allowUnfree = true; }
               in configuration.nix to override this.
               Alternatively you can configure a predicate to allow specific packages:
                 { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
                     "slack"
                   ];
                 }
               c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
                 { allowUnfree = true; }
               to ~/.config/nixpkgs/config.nix.
        (use '--show-trace' to show detailed location information)
[1/1] slack: Fail
There was an error installing nix packages: slack. Packages were not added to devbox.json
Error: Command: /nix/var/nix/profiles/default/bin/nix profile install --profile /Users/savil/code/jetpack/devbox/examples/testdata/unfree-packages/.devbox/nix/profile/default nixpkgs/9e0eed654c705c7cafe192a8eba1610217f70544#slack --extra-experimental-features ca-derivations --option experimental-features nix-command flakes --priority 5: exit status 1

Error: There was an internal error. Run with DEVBOX_DEBUG=1 for a detailed error message, and consider reporting it at https://github.com/jetpack-io/devbox/issues
```

But we can add this back, and I think it won't hurt the shell startup time, no?